src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -0,0 +1,150 @@
+package wolox.training.controllers;
+
+import java.util.List;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.exceptions.UserNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.models.UserBook;
+import wolox.training.repositories.BookRepository;
+import wolox.training.repositories.UserRepository;
+
+
+/**
+ * This class permits create, update, deleted and list over User object
+ *
+ * @author carolina.marulanda@wolox.co
+ */
+@RestController
+@RequestMapping("/users")
+public class UserController {
+
+    /**
+     * Injects UserRepository repository for user's operations
+     */
+    @Autowired
+    private UserRepository userRepository;
+
+    /**
+     * Injects BookRepository repository for book's operations
+     */
+    @Autowired
+    private BookRepository bookRepository;
+
+
+    /**
+     * This method returns a user {@link UserBook}, which match with username received as parameter
+     *
+     * @param username user's username
+     *
+     * @return {@link Optional} book's user
+     */
+    @GetMapping("/user/{userUsername}")
+    public Optional<UserBook> findByUsername(@PathVariable String username) {
```

Comment:
We shouldn't return an Optional, remove the wrapper and use something like userRepository.findByUsername(username).orElseThrow(...)

Reply:
fixed